### PR TITLE
WIP: Bug 2064741  purge mounting fs cgroup

### PR DESCRIPTION
**Pull request by harche**
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by harche** [comment]
/hold

Created for testing with cluster-bot

File: server/container_create_linux.go [hunk 1]
```diff
@@ -1076,18 +1076,32 @@ func setupSystemd(mounts []rspec.Mount, g generate.Generator) {
...
 		}
-		g.AddMount(systemdMnt)
 		g.AddLinuxMaskedPaths("/sys/fs/cgroup/systemd/release_agent")
 	}
 	g.AddProcessEnv("container", "crio")
 }
 
+func doesntHaveSysFsCgroupFromHost(mounts []rspec.Mount) bool {
+	for _, m := range mounts {
+		if m.Destination == "/sys/fs/cgroup" {
```
Comment (haircommander) [inline server/container_create_linux.go:1098]: does this PR work if the user manually mounts `/sys`? i.e. we currently skip if the direct parent is mounted, but don't check any indirect. should we?

**Comment by haircommander** [comment]
/ok-to-test
/approve

I think this approach is the one we want, so we don't risk systemd not being able to access something  because the mount is ro

BTW, you've got a lint issue

**Comment by harche** [comment]
/close 

We have decided to go with https://github.com/cri-o/cri-o/pull/5778

**Pull request closed without merging**